notificationservice.cpp

```diff
@@ -25,9 +26,9 @@ NotificationData NotificationService::question() const {
 }
 
 void NotificationService::setNotify(const NotificationData& notify) {
-    if (_notify != notify)
-        _history.push_back(_notify);
-
+    if (_notify != notify) {
+        _history->addHistoryObject(notify);
```

Comment:
I think your code will be crashed here. Because you still not initialize your _history object... 
please add initialization into constructor of the NotificationService